Remove python-igraph--specific steps from CI and README

ulupo opened the pull request:
#### Reference Issues/PRs
(#251), (#206)


#### What does this implement/fix? Explain your changes.
FIxes (#206) following the release of ``python-igraph`` v0.8.0 a couple of days ago, now including wheels for all systems. The CI and installation instructions no longer need to treat this dependency differently from the other ones.

Some unrelated linting was also performed in test files, as well as fixes to silence warnings in `diagrams._utils.py` and in `setup.cfg` (for `pytest`).

Finally, a test which we had not been noticed was failing was fixed in `gtda/mapper/test_visualization.py`.


#### Any other comments?
The minimum version requirements for ``python-igraph`` is still ``0.7.1post6``. ``pycairo`` no longer needs to be installed separately on Windows.

Commit:
Remove python-igraph--specific steps from CI

Commit:
Remove unnecessary file

Commit:
Silence junit_family pytest warning correctly

Commit:
Linting

Commit:
Silence NumPy stack warning from _bin

Commit:
Remove notes on python-igraph installation from README.rst

Commit:
Merge branch 'master' into dep/python-igraph

lewtun reviewed (approved):
Looks good to me!

Commit:
Merge branch 'master' into dep/python-igraph

wreise committed:
Change the hovertext test

wreise commented on azure-pipelines.yml:
```diff
@@ -130,13 +130,6 @@ jobs:
     failOnStderr: true
     displayName: 'Upgrade pip and setuptools'
 
-  - bash: |
-      pycairo_url=$(python .azure-ci/python-igraph_dependencies_win.py '$(python_ver)' 'pycairo')
-      igraph_url=$(python .azure-ci/python-igraph_dependencies_win.py '$(python_ver)' 'python-igraph')
-      pip install "$pycairo_url" "$igraph_url"
-    failOnStderr: true
-    displayName: 'Install pycairo and python-igraph'
-
```
Just to confirm, python-igraph is now installed as a normal dependency?

ulupo replied:
That's precisely the point of the PR, yes :)

ulupo committed:
Merge pull request #12 from wreise/w-p-igraph-dep

Change the hovertext test

Commit:
Add empty init file in mapper test folder

Attempts to solve discrepancy whereby mapper tests seem not to be performed in wheel testing part of Azure CI
